sphinx/writers/latex.py

```diff
@@ -792,26 +792,24 @@ def generate_numfig_format(self, builder):
             ret.append('\\def\\fnum@figure{%s}\n' %
                        text_type(figure[0]).strip().translate(tex_escape_map))
         else:
-            definition = text_type(figure[0]).strip().translate(tex_escape_map)
+            definition = escape_abbr(text_type(figure[0]).translate(tex_escape_map))
             ret.append(self.babel_renewcommand('\\figurename', definition))
-            if figure[1]:
-                ret.append('\\makeatletter\n')
-                ret.append('\\def\\fnum@figure{\\figurename\\thefigure%s}\n' %
-                           text_type(figure[1]).strip().translate(tex_escape_map))
-                ret.append('\\makeatother\n')
+            ret.append('\\makeatletter\n')
+            ret.append('\\def\\fnum@figure{\\figurename \\thefigure{}%s}\n' %
+                       text_type(figure[1]).translate(tex_escape_map))
+            ret.append('\\makeatother\n')
 
         table = self.builder.config.numfig_format['table'].split('%s', 1)
         if len(table) == 1:
             ret.append('\\def\\fnum@table{%s}\n' %
                        text_type(table[0]).strip().translate(tex_escape_map))
         else:
-            definition = text_type(table[0]).strip().translate(tex_escape_map)
+            definition = escape_abbr(text_type(table[0]).translate(tex_escape_map))
             ret.append(self.babel_renewcommand('\\tablename', definition))
-            if table[1]:
-                ret.append('\\makeatletter\n')
-                ret.append('\\def\\fnum@table{\\tablename\\thetable%s}\n' %
-                           text_type(table[1]).strip().translate(tex_escape_map))
-                ret.append('\\makeatother\n')
+            ret.append('\\makeatletter\n')
+            ret.append('\\def\\fnum@table{\\tablename \\thetable{}%s}\n' %
```

Comment:
Here too, there is no reason for the added space after `\\tablename` I will remove.